canvases/dark-tree-sitter-demo/main.dark

```diff
@@ -0,0 +1,377 @@
+[<HttpHandler("GET", "/static/:path")>]
+let _handler _req =
+  let body =
+    (Experiments.readFromStaticDir path)
+    |> Result.withDefault Bytes.empty
+
+  let contentType =
+    match (String.split_v1 path ".") |> List.last_v2 |> Option.withDefault "" with
+    | "wasm" -> "application/wasm"
+    | "js" -> "text/javascript"
+    | "dll" -> "application/octet-stream"
+    | "dat" -> "application/octet-stream"
+    | "blat" -> "application/octet-stream"
+    | "json" -> "application/json"
+    | _ -> "text/plain"
+
+  let incomingHost =
+    request.headers
+    |> List.findFirst_v2 (fun t -> Tuple2.first t == "origin")
+    |> Option.map_v1 (fun t -> Tuple2.second t)
+    |> Option.withDefault "Couldn't find 'origin' request header to determine Access-Control-Allow-Origin"
+
+  let headers =
+    Dict.fromListOverwritingDuplicates
+      [
+        ("Access-Control-Allow-Origin", incomingHost)
+        ("Access-Control-Allow-Credentials", "true")
+        ("Content-type", contentType)
+        ("Accept-Ranges", "bytes")
+      ]
+
+  Http.responseWithHeaders body headers 200
+
+[<HttpHandler("GET", "/")>]
+let _indexHandler _req =
+  // TODO: HTML of a basic webpage, that references WASM stuff
+  let body =
+    """
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Dark Tree-Sitter WASM demo</title>
+    <style>
+      * {font-size: 24px;}
+      body, html { height: 100%; margin: 0; font-family: Arial, sans-serif }
+      .container { display: flex; height: 100%; }
+      .left-side, .middle, .right-side { flex: 1; padding: 10px; }
+      .textarea { width: 100%; height: 95%; resize: none; }
+    </style>
+  </head>
+
+  <body>
+    <div class="container">
+      <div class="left-side">
+        <label>Source code:</label>
+        <textarea id="input-textarea" class="textarea">let add (a: Int) (b: Int): Int =
+  let sum = a + b
+  sum</textarea>
+      </div>
+
+      <div class="middle">
+        <label>Syntax tree</label>
+        <textarea id="syntax-textarea" class="textarea" disabled></textarea>
+      </div>
+
+      <div class="right-side">
+        <label>Vanilla-serialized JSON of PT.UserFunction</label>
+        <textarea id="right-textarea" class="textarea" disabled></textarea>
+      </div>
+    </div>
+    <script src="/static/tree-sitter/tree-sitter.js" />
+    <script>console.log('NOT SURE WHY I NEED THIS')</script>
```

Comment:
TODO: I need to check if I still need this - for some reason, the tree-sitter bindings seemed to fail unless I added an extra script tag above the main one.